pkg/kubectl/cmd/util/factory.go

```diff
@@ -145,6 +145,9 @@ type Factory struct {
 	// UpdatePodSpecForObject will call the provided function on the pod spec this object supports,
 	// return false if no pod spec is supported, or return an error.
 	UpdatePodSpecForObject func(obj runtime.Object, fn func(*api.PodSpec) error) (bool, error)
+	// UpdateSelectorForObject calls fn with that applicable selector type to update.
+	UpdateSelectorForObject func(obj runtime.Object, fn func(*map[string]string, *unversioned.LabelSelector) error) error
```

Comment:
I should be able to co-erce every map based selector into an `unversioned.LabelSelector` and simply use that, right?  Types that only support one can validate that the set is legal.